themes/src/main/resources/theme/base/login/resources/js/kcMultivalued.js

```diff
@@ -0,0 +1,101 @@
+const DATA_KC_MULTIVALUED = 'data-kcMultivalued';
+const KC_ADD_ACTION_PREFIX = "kc-add-";
+const KC_REMOVE_ACTION_PREFIX = "kc-remove-";
+
+function createAddAction(element) {
+    const name = element.getAttribute("name");
+    const action = document.createElement("a")
+    action.setAttribute("id", KC_ADD_ACTION_PREFIX + element.getAttribute("id"));
+    action.setAttribute("href", "#");
+    action.innerText = "Add " + window.keycloak.profile.getAttribute(name).getDisplayName();
```

Comment:
Global objects such as `window` should not be used to hold state. If you need this information to be shared between different pieces of JS they should use a shared module.

Reply:
How to dynamically build a module based on the user profile metadata without introducing changes to our backend?

Reply:
I've removed this from the PR so we can enable exposing metadata to user-defined modules as a follow-up.